recipes/ixwebsocket/all/conanfile.py

```diff
@@ -0,0 +1,104 @@
+from conans import ConanFile, CMake, tools
+from conans.tools import Version
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class IXWebSocketConan(ConanFile):
+    name = "ixwebsocket"
+    description = "IXWebSocket is a C++ library for WebSocket client and server development"
+    topics = ("conan", "IXWebSocket", "socket", "websocket")
+
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/machinezone/IXWebSocket"
+
+    # License for the library
+    license = "BSD-3-Clause"
+
+    exports_sources = ["CMakeLists.txt"]
+
+    settings = "os", "compiler", "build_type", "arch"
+
+    short_paths = True
+    generators = "cmake"
+
+    options = {
+        "use_mbed_tls": [False, True],
+        "use_tls": [True, False],
+        "use_openssl": [False, True],
+        "fPIC": [True, False]
+    }
+    default_options = {k: v[0] for k, v in options.items()}
+
+    def _can_use_openssl(self):
+        if self.settings.os == "Windows":
+            # Future: support for OpenSSL on Windows was introduced in 7.9.3. Earlier versions force MbedTLS
+            return Version(self.version) >= "7.9.3"
+        # The others do, by default, support OpenSSL and MbedTLS. Non-standard operating systems might
+        # be a challenge.
+        # Older versions doesn't support OpenSSL on Mac, but those are unlikely to be built now.
+        return True
+
+    def configure(self):
+        if self.options.use_mbed_tls and not self.options.use_tls or self.options.get_safe("use_openssl") and not self.options.use_tls:
+            raise ConanInvalidConfiguration("TLS must be enabled to use mbedtls")
+        elif self.options.use_mbed_tls and self.options.get_safe("use_openssl"):
+            raise ConanInvalidConfiguration("Cannot use both OpenSSL and MbedTLS")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            self.default_options["use_mbed_tls"] = True
+            if not self._can_use_openssl():
+                del self.options.use_openssl
+        elif self.settings.os == "Linux":
+            # On Linux, if TLS is enabled and mbed is disabled, the only option is OpenSSL.
+            del self.options.use_openssl
+
+    def requirements(self):
+
+        if(self._can_use_openssl() and not self.options.use_mbed_tls and self.options.use_tls
+           and ("use_openssl" not in self.options or self.options.use_openssl)):
```

Comment:
```suggestion
           and (self.options.get_safe("use_openssl")):
```

Reply:
that's not entirely the same though. It's an `or`, because `use_openssl` is removed on Linux. if get_safe returns false if it's not present, it breaks the check on Linux

Reply:
What does mean, on Linux, OpenSSL is mandatory, that's why use_openssl is removed when settings.os is Linux?

Reply:
OpenSSL is a part of an else statement (https://github.com/machinezone/IXWebSocket/blob/v7.6.3/CMakeLists.txt#L120), which only triggers on Linux if mbedtls is disabled. Basically, OpenSSL is forced on Linux. That's why I removed the key too, because it saves trouble with the fallback.

The rest of the if-statement checks for the validity. It only triggers if mbed is disabled, tls is enabled, and OpenSSL is enabled or this is Linux.

Reply:
I removed it because it's not an option. If you use TLS and run Linux, and don't use mbed, you get OpenSSL. That being said, for future update purposes, checking os != Mac should account for 7.9.5. It's also more flexible than a rigid Linux check.

Reply:
Thanks for your detailed explanation